tests/integration/security/sds_tls_origination/util/util.go

```diff
@@ -0,0 +1,410 @@
+	tlsScrtCert = "tls.crt"
+	// The ID/name for the k8sKey in kubernetes tls secret.
+	tlsScrtKey = "tls.key"
+	// The ID/name for the CA certificate in kubernetes tls secret
+	tlsScrtCaCert = "ca.crt"
+	// The ID/name for the certificate chain in kubernetes generic secret.
+	genericScrtCert = "cert"
+	// The ID/name for the private key in kubernetes generic secret.
+	genericScrtKey = "key"
+	// The ID/name for the CA certificate in kubernetes generic secret.
+	genericScrtCaCert = "cacert"
+)
+
+// CreateKubeSecret reads credential names from credNames and key/cert from TLSCredential,
+// and creates K8s secrets for gateway.
+// nolint: interfacer
+func CreateKubeSecret(t test.Failer, ctx framework.TestContext, credNames []string,
+	credentialType string, egressCred TLSCredential, isNotGeneric bool) {
+	t.Helper()
+	// Get namespace for gateway pod.
+	istioCfg := istio.DefaultConfigOrFail(t, ctx)
+	systemNS := namespace.ClaimOrFail(t, ctx, istioCfg.SystemNamespace)
+
+	if len(credNames) == 0 {
+		ctx.Log("no credential names are specified, skip creating secret")
+		return
+	}
+	// Create Kubernetes secret for gateway
+	cluster := ctx.Environment().(*kube.Environment).KubeClusters[0]
+	for _, cn := range credNames {
+		secret := createSecret(credentialType, cn, systemNS.Name(), egressCred, isNotGeneric)
+		_, err := cluster.CoreV1().Secrets(systemNS.Name()).Create(context.TODO(), secret, metav1.CreateOptions{})
+		if err != nil {
+			t.Fatalf("Failed to create secret (error: %s)", err)
+		}
+	}
+	// Check if Kubernetes secret is ready
+	retry.UntilSuccessOrFail(t, func() error {
+		for _, cn := range credNames {
+			_, err := cluster.CoreV1().Secrets(systemNS.Name()).Get(context.TODO(), cn, metav1.GetOptions{})
+			if err != nil {
+				return fmt.Errorf("secret %v not found: %v", cn, err)
+			}
+		}
+		return nil
+	}, retry.Timeout(time.Second*5))
+}
+
+// DeleteKubeSecret deletes a secret
+// nolint: interfacer
+func DeleteKubeSecret(t test.Failer, ctx framework.TestContext, credNames []string) {
+	// Get namespace for gateway pod.
+	istioCfg := istio.DefaultConfigOrFail(t, ctx)
+	systemNS := namespace.ClaimOrFail(t, ctx, istioCfg.SystemNamespace)
+
+	if len(credNames) == 0 {
+		ctx.Log("no credential names are specified, skip creating ingress secret")
+		return
+	}
+	// Create Kubernetes secret for ingress gateway
+	cluster := ctx.Environment().(*kube.Environment).KubeClusters[0]
+	for _, cn := range credNames {
+		var immediate int64
+		err := cluster.CoreV1().Secrets(systemNS.Name()).Delete(context.TODO(), cn,
+			metav1.DeleteOptions{GracePeriodSeconds: &immediate})
+		if err != nil {
+			t.Fatalf("Failed to create secret (error: %s)", err)
+		}
+	}
+}
+
+// createSecret creates a kubernetes secret which stores CA cert for SIMPLE TLS.
+// For mTLS ingress gateway, createSecret adds client cert and key into the secret object.
+func createSecret(credentialType string, cn, ns string, ic TLSCredential, isNotGeneric bool) *v1.Secret {
```

Comment:
Could you revise the util function createSecret() in tests/integration/security/sds_ingress/util/util.go to avoid adding a similar util function here?

Reply:
The way it creates secret is different as it doesn't expect to create CA cert in SIMPLE TLS mode, I'll see if this can be refactored

Reply:
createSecret creates secret data of type []byte whereas data of type string is required for this test as we read certs from files too. Moreover createSecret doesn't know how to create CA only secrets. This makes it hard to refactor both into a single function